instantsearch/src/main/java/com/algolia/instantsearch/helpers/Searcher.java

```diff
@@ -448,35 +451,43 @@ public Searcher addNumericRefinement(@NonNull NumericRefinement refinement) {
      */
     @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
     public Searcher removeNumericRefinement(@NonNull String attribute) {
-        numericRefinements.remove(attribute);
-        rebuildQueryNumericFilters();
-        return this;
+        bus.post(new NumericRefinementEvent(Operation.REMOVE, new NumericRefinement(attribute, -42, -42)));
+        return removeNumericRefinement(attribute, null, false);
     }
 
 
     /**
-     * Removes the numeric refinement relative to an attribute and operator for the next queries.
+     * Removes the given numeric refinement for the next queries.
      *
-     * @param attribute an attribute that maybe has some refinements.
-     * @param operator  an {@link NumericRefinement#OPERATOR_EQ operator}.
+     * @param refinement a description of the refinement to remove.
      */
     @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
-    public Searcher removeNumericRefinement(@NonNull String attribute, int operator) {
-        NumericRefinement.checkOperatorIsValid(operator);
-        numericRefinements.get(attribute).remove(operator);
-        rebuildQueryNumericFilters();
-        return this;
+    public Searcher removeNumericRefinement(@NonNull NumericRefinement refinement) {
+        bus.post(new NumericRefinementEvent(Operation.REMOVE, refinement));
+        return removeNumericRefinement(refinement.attribute, refinement.operator, false);
     }
 
     /**
-     * Removes the given numeric refinement for the next queries.
+     * Removes the numeric refinement relative to an attribute and operator for the next queries.
      *
-     * @param refinement a description of the refinement to remove.
+     * @param attribute an attribute that maybe has some refinements.
+     * @param operator  an {@link NumericRefinement#OPERATOR_EQ operator}.
      */
     @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
-    public Searcher removeNumericRefinement(@NonNull NumericRefinement refinement) {
-        NumericRefinement.checkOperatorIsValid(refinement.operator);
-        numericRefinements.get(refinement.attribute).remove(refinement.operator);
+    public Searcher removeNumericRefinement(@NonNull String attribute, @Nullable Integer operator) {
+        return removeNumericRefinement(attribute, operator, true);
+    }
+
+    private Searcher removeNumericRefinement(@NonNull String attribute, @Nullable Integer operator, boolean sendEvent) {
```

Comment:
concerning the `sendEvent` boolean:
Is there a case where you do not want to send an event when a parameter is removed? Because I think that we always want to send an event, so I suggest you do the following:
1- remove the `bus.post(new NumericRefinementEvent(Operation.REMOVE...` line in the 2 functions above
2- remove the sendEvent parameter and the conditional statement that uses it below.